File: myExpenses/src/main/java/org/totschnig/myexpenses/ui/ContextAwareRecyclerView.java
package org.totschnig.myexpenses.ui;

import android.content.Context;
import android.util.AttributeSet;
import android.view.ContextMenu;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

public class ContextAwareRecyclerView extends RecyclerView {
  private RecyclerContextMenuInfo contextMenuInfo;

  public ContextAwareRecyclerView(Context context) {
    super(context);
  }

  public ContextAwareRecyclerView(Context context, @Nullable AttributeSet attrs) {
    super(context, attrs);
  }

  public ContextAwareRecyclerView(Context context, @Nullable AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
  }

  @Override
  protected RecyclerContextMenuInfo getContextMenuInfo() {
    return contextMenuInfo;
  }

  @Override
  public boolean showContextMenuForChild(View originalView) {
    final int position = getChildAdapterPosition(originalView);
    if (position >= 0) {
      contextMenuInfo = new RecyclerContextMenuInfo(position, getChildViewHolder(originalView).getItemId());
      return super.showContextMenuForChild(originalView);
    }
    return false;
  }

  public class RecyclerContextMenuInfo implements ContextMenu.ContextMenuInfo {

    public RecyclerContextMenuInfo(int position, long id) {
      this.position = position;
      this.id = id;
    }

    final public int position;
    final public long id;
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/ui/ContextAwareRecyclerView.kt
package org.totschnig.myexpenses.ui

import android.content.Context
import android.util.AttributeSet
import android.view.ContextMenu.ContextMenuInfo
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class ContextAwareRecyclerView : RecyclerView {
    private var contextMenuInfo: RecyclerContextMenuInfo? = null

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) :
            super(context, attrs, defStyle)

    override fun getContextMenuInfo(): RecyclerContextMenuInfo? {
        return contextMenuInfo
    }

    override fun showContextMenuForChild(originalView: View): Boolean {
        val position = getChildAdapterPosition(originalView)
        if (position >= 0) {
            contextMenuInfo =
                RecyclerContextMenuInfo(position, getChildViewHolder(originalView).itemId)
            return super.showContextMenuForChild(originalView)
        }
        return false
    }

    class RecyclerContextMenuInfo(val position: Int, val id: Long) : ContextMenuInfo
}
